systemtap/PKGBUILD

```diff
@@ -2,30 +2,30 @@
 # Contributor : Christian Rebischke <Chris.Rebischke@archlinux.org>
 # Contributor : dront78 <dront78@gmail.com>
 pkgname=systemtap
-pkgver=3.3
+pkgver=4.0
 pkgrel=1
 pkgdesc="provides infrastructure to simplify the gathering of information about the running system."
 url="http://sourceware.org/systemtap/"
 arch=('x86_64' 'i686')
 license=('GPL')
-depends=('elfutils' 'nss' 'python2' 'json-c')
-makedepends=('python2-setuptools' 'xmlto')
+depends=('elfutils' 'nss' 'python2' 'json-c' 'avahi')
+makedepends=('python2-setuptools' 'xmlto' 'cpio')
 optdepends=('sqlite3: for storing results in a database')
 source=("${pkgname}-${pkgver}.tar.gz::https://sourceware.org/systemtap/ftp/releases/${pkgname}-${pkgver}.tar.gz"
         "${pkgname}-${pkgver}.tar.gz.asc::https://sourceware.org/systemtap/ftp/releases/${pkgname}-${pkgver}.tar.gz.asc"
         )
-sha512sums=('b75a4591bdc021645c15cb8f2b8991f46fdffb29b1d132745bafe4291aee5e1892ea9a63c8e98f011a4fee68decd99aa4401dc2f70e163e801cd140ad4cd6b6e'
+sha512sums=('66b5ba1902ed974101429dce9fee849273d3dc64e862c057177d158c5f284cc93299819ee8ee49d4bd3b7ff47a3512b1c68b74cad374147ce67fa406c7518c79'
             'SKIP')
 install='systemtap.install'
-validpgpkeys=('41A0C11274B1E87FE28E4C53E3D77E1F95161991')
+validpgpkeys=('5D38116FA4D3A7CC77E378D37E83610126DCC2E8')
 
 
 build() {
   cd "${pkgname}-${pkgver}"
   sed -i -e "42,50d" stapbpf/Makefile.am
-  sed -i -e "840,848d" stapbpf/Makefile.in
-  sed -i -e "73,80d" staprun/Makefile.am
-  sed -i -e "1110,1117d" staprun/Makefile.in
+  sed -i -e "842,850d" stapbpf/Makefile.in
+  sed -i -e "77,84d" staprun/Makefile.am
+  sed -i -e "1115,1122d" staprun/Makefile.in
```

Comment:
Using line numbers is fragile and hard to maintain. Patches should be better solution.

By the way, [PKGBUILD for the same package on AUR](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=systemtap) does not have such lines. Are they necessary?

Reply:
IIUIC these lines will be executed as the `package()` function in PKGBUILD is run under fakeroot, so `id -u` return 0. In Arch Linux, it's encouraged to use `sysusers.d` instead of adding users and groups manually [1], so it's better to remove these lines from Makefiles.

[1] https://www.archlinux.org/todo/switch-to-systemd-sysusers/

Reply:
Yes you are right, I tried with fakeroot `id -u` return 0. The removed lines in Makefile actually will take effect in `package()`. Sorry for my misunderstanding.

Reply:
I checked [this][1], is below a better way ?

1. do not remove lines in Makefiles, thus mode bits and group of `staprun` `stapbpf` will be already ok in generated package
2. add a systemtap.sysusers
3. remove group adding, chgrp, chmod in systemtap.install

Is this ok? not sure about that.

[1]: https://www.archlinux.org/todo/switch-to-systemd-sysusers/

Reply:
Oops I didn't notice there are `chown` commands. In this it might be better to keep the current approach and not moving to sysusers.

Reply:
I've already changed. Do i need to close this PR and recreate another?

Reply:
You can just continue your work in this PR.

Let's wait for lijinpei before further decisions.